docs/reference/search/search_context.asciidoc

```diff
@@ -0,0 +1,92 @@
+[[search-context]]
+==== Search Context
+
+By default, a search request executes against the latest point in time readers of the
+target indices. Sometimes it's more preferred to execute multiple search requests using
+the same point in time readers. For example, the combined result of the initial and
+subsequent search_after requests would be more consistent if they use the same point
+in time readers. This can be done by using a single search context for all search requests.
+
+A search context must be opened in a separate step before being used in subsequent
+search requests. The keep_alive parameter tells Elasticsearch how long it should keep
+the seach context alive, e.g. `?keep_alive=5m`.
+
+[source,console]
+--------------------------------------------------
+POST /twitter/_search_context?keep_alive=1m
+--------------------------------------------------
+// TEST[setup:twitter]
+
+The result from the above request includes a `id`, which should
+be passed to the `id` of the `search_context` parameter of a search request.
+
+POST /_search <1>
+{
+    "size": 100,
+    "query": {
+        "match" : {
+            "title" : "elasticsearch"
+        }
+    },
+    "search_context": {
+	    "id":  "46ToAwEHdHdpdHRlchZSa1dRZDFELVNLbTZkQVJudUtGMFFnAAAWNVAwS09JWTdTRUdibWE1ZC1id0tjQRRURnhPWjNFQmI3bmVOeVZReS1tRAAAAAAAAAAB", <2>
+	    "keep_alive": "1m"  <3>
+    }
+}
+
+<1> A search request with `search_context` must not specify `index`, `routing`,
+and `preferences` as these parameters are copied from the `search_context`.
+<2> The `id` parameter tells Elasticsearch to execute the request using
+the point-in-time readers from this search context id.
+<3> The `keep_alive` parameter tells Elasticsearch how long it should extend
+the time to live of the search context
+
+IMPORTANT: The open search context request and each subsequent search request can
+return different `id`; thus always use the most recently received `id` for the
```

Comment:
I am interested in what situations search context id can be different between requests?  I assume if all the original shards are still available, we expect search context is to be the same?

Reply:
I think we should remove this disclaimer. We don't change the `reader_id` today but I think this was added to copy scrolls which have the same warning. That leaves the door open in case we want to change the scroll id in the future. However this is different than scrolls here since the `reader_id` can be shared so I don't think this should be left open.
@dnhatn wdyt ?

Reply:
Yeah, that was my intention. When we implement the retry logic (i.e., try with other shards when the target shards are not available), the id will change. I am okay to remove it now if it's confusing.